rs/consensus/src/idkg/utils.rs

```diff
@@ -443,6 +443,18 @@ pub(crate) fn algorithm_for_key_id(key_id: &MasterPublicKeyId) -> AlgorithmId {
             SchnorrAlgorithm::Bip340Secp256k1 => AlgorithmId::ThresholdSchnorrBip340,
             SchnorrAlgorithm::Ed25519 => AlgorithmId::ThresholdEd25519,
         },
+        MasterPublicKeyId::VetKd(_vetkd_key_id) => {
+            todo!(
+                "how to behave here: it seems `algorithm_for_key_id` is only
+                called once in production by fn update_next_key_transcript in
+                the iDKG payload builder. Seemingly all other of the many
+                usages usages are in test code.
+                "
+            )
+            // match vetkd_key_id.curve {
+            //     VetKdCurve::Bls12_381 => AlgorithmId::ThresBls12_381,
+            // },
+        }
```

Comment:
I agree, also even if we don't use it, its best if the function simply does what it says.

Reply:
`algorithm_for_key_id` (in `rs/consensus/src/idkg/utils.rs`) now returns an algorithm:
```
[...]
    MasterPublicKeyId::VetKd(vetkd_key_id) => match vetkd_key_id.curve {
        VetKdCurve::Bls12_381 => AlgorithmId::ThresBls12_381,
    },
[...]
```
I'm not fully sure yet if we can reuse the existing `AlgorithmId::ThresBls12_381` or if we should introduce a new dedicated algorithm. Reusing the existing _seems_ fine as long as there are no collisions with other master public key algorithms, especially since (unless I'm missing something) we don't (and don't intend to) store this algorithm ID anywhere (e.g., in the registry or in a secret key store). WDYT?

If you want me to add some specific items/reminders to the CON-1412 that you mentioned, please let me know.

Reply:
If I remember correctly from yesterday's discussion, we agreed to use the `AlgorithmId::Placeholder` because the algorithm ID for vetKD is meaningless for now because neither the CryptoComponent APIs will use an algorithm ID nor will this algorithm be persisted anywhere (e.g., in the registry or in a key store).

I adapted the `algorithm_for_key_id` accordingly. With this, I'm actually not sure if it still makes sense to return an algorithm for vetKD at all in `algorithm_for_key_id`. Please let me know if you want me to remove the line.

Reply:
I think this is fine for now. Alternatively we can change the signature of `algorithm_for_key_id` to return a `Result` which we can then handle properly in non-test code. But we can take a closer look at that as part of CON-1412.

Reply:
Ok, I leave it as is then (i.e., return `AlgorithmId::Placeholder` for the vetKD variant) and let you handle it in CON-1412. With this, I'm resolving the conversation.